### PR TITLE
[DASA-2822][DSVA-294] - Fix training loop - remove fast cocoeval api, add mlflow to training and evaluation loop and update requirements.

### PR DESCRIPTION
**Pull request by beazy96**
Mlflow experiment: http://192.168.189.67:8888/#/experiments/388

### CODE REVIEW

File: yolox/evaluators/coco_evaluator.py [hunk 1]
```diff
@@ -254,7 +255,7 @@ def convert_to_coco_format(self, outputs, info_imgs, ids, return_outputs=False):
 
     def evaluate_prediction(self, data_dict, statistics):
         if not is_main_process():
-            return 0, 0, None
+            return 0, 0, None, {0: 0}, {0: 0}
```
Comment (pkolebski) [inline yolox/evaluators/coco_evaluator.py:258]: return empty dicts

File: yolox/evaluators/coco_evaluator.py [hunk 2]
```diff
@@ -290,12 +291,9 @@ def evaluate_prediction(self, data_dict, statistics):
...
                 json.dump(data_dict, open(tmp, "w"))
                 cocoDt = cocoGt.loadRes(tmp)
-            try:
-                from yolox.layers import COCOeval_opt as COCOeval
-            except ImportError:
-                from pycocotools.cocoeval import COCOeval
 
-                logger.warning("Use standard COCOeval.")
+            from pycocotools.cocoeval import COCOeval
+            logger.warning("Use standard COCOeval.")
```
Comment (pkolebski) [inline yolox/evaluators/coco_evaluator.py:296]: remove log as this is only available option now.

File: yolox/layers/__init__.py [hunk 3]
```diff
@@ -1,13 +1 @@
...
-# import torch first to make jit op work without `ImportError of libc10.so`
-import torch  # noqa
-
-from .jit_ops import FastCOCOEvalOp, JitOp
-
-try:
-    from .fast_coco_eval_api import COCOeval_opt
-except ImportError:  #  exception will be raised when users build yolox from source
-    pass
+"""Module deleted."""
```
Comment (pkolebski) [inline yolox/layers/__init__.py:1]: just delete this file if it is the last one in directory. If not, change docstring to something more informative.

File: yolox/layers/fast_coco_eval_api.py [hunk 4]
```diff
@@ -1,151 +1 @@
...
-        # num_area_ranges X num_max_detections
-        self.eval["precision"] = np.array(self.eval["precision"]).reshape(
-            self.eval["counts"]
-        )
-        self.eval["scores"] = np.array(self.eval["scores"]).reshape(self.eval["counts"])
-        toc = time.time()
-        print(
-            "COCOeval_opt.accumulate() finished in {:0.2f} seconds.".format(toc - tic)
-        )
+"""Module deleted."""
```
Comment (pkolebski) [inline yolox/layers/fast_coco_eval_api.py:1]: delete this file.

**Review by pkolebski — CHANGES_REQUESTED** [review]

File: yolox/layers/jit_ops.py [hunk 5]
```diff
@@ -84,7 +84,7 @@ def load(self, verbose=True):
         try:
             # try to import op from pre-installed package
             return importlib.import_module(self.absolute_name())
-        except Exception:  # op not compiled, jit load
+        except (Exception, ModuleNotFoundError):  # op not compiled, jit load
```
Comment (dpolak-dtiq) [inline yolox/layers/jit_ops.py:87]: remove this inline comment

**Review by dpolak-dtiq — CHANGES_REQUESTED** [review]

**Review by pkolebski — APPROVED** [review]

**Review by dpolak-dtiq — APPROVED** [review]

**Pull request merged**